### DISCUSSION TITLE
Aeris forecast.json not being created

### DISCUSSION THREAD

**Discussion by averagejoej** [poblabs/weewx-belchertown#751] [Q&A] [answered]
I recently switched over to this weewx skin, and have been working on setting up all the cool features (MQTT updates are awesome), but I've run into some trouble with the forecast.json being created.

I run weewx inside of a docker container, so my initial thought was that the container wasn't able to access the Aeris API, but I was able to validate one of the API calls to Aeris with curl inside the container.

debug mode inside a browser doesn't provide much other than the file not being found:
`/belchertown/json/fo…n?_=1641263687086:1 Failed to load resource: the server responded with a status of 404 (Not Found)`

Looking at the syslog, I don't see the exception being raised from line [971](https://github.com/poblabs/weewx-belchertown/blob/57934682d4dca735fe65e3985a311bfb7e5dfb23/bin/user/belchertown.py#L971) (v 1.2) so maybe it's not getting that far?

Anyways, I'm really just looking for some ideas on how to troubleshoot this further, as opposed to creating an issue right away.  I'm not super familiar with how weewx works under the hood, so I'm not really sure where to go from here.  Any help is much appreciated.  Unfortunately, my site is not exposed publicly - but I'm not sure what anyone could see from there anyways.

thanks!
Joe

**Comment by michaelundwd** [comment, marked as answer]
Here's some hints that might help you track this down.

1. From your evidence, it's not immediately clear whether forecast.json has been generated by weewx/belchertown or is not in the right place on your server. 
2. The forecast.json file is generated by weewx/belchertown once an hour and it leaves a record in syslog, ("New forecast file downloaded to /var/weewx/reports/live/json/forecast.json" on my system), so it's worth checking that you are getting such a message once an hour.
3. To check if it's in the right place, check the contents of the json file on your server.  If all's well, it should be there along with json files for any graphs you may have set up.
4. Make sure your forecast settings are correct - see the forecast options in the readme.md file on the belchertown github page
5. It's a good idea to restart weewx after changing the forecast settings so that the changes are taken up in the next weewx cycle, then wait for a couple of cycles for it to settle down.

HTH

Michael

**Reply by averagejoej** [reply]
Thanks for the reply.  Turned out it was a really dumb typo in weewx.conf.  One run of wee_reports and forecast.json appeared!

![image](https://user-images.githubusercontent.com/12356777/148410753-cb06c90e-869f-4fc3-8907-c9c9d4f44e57.png)
